src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -2178,25 +2304,98 @@ create_hw_context(const xrt::uuid& xclbin_uuid,
                   const xrt::hw_context::qos_type& qos,
                   xrt::hw_context::access_mode mode)
 {
-  // Explicit hardware contexts are not supported in Alveo.
-  throw xrt_core::ishim::not_supported_error{__func__};
+  if (!hw_context_enable) {
+    // Nothing to be done here for legacy flow. Return from here
+    return 0;
```

Comment:
The legacy implementation was to throw `throw xrt_core::ishim::not_supported_error{__func__};` maybe you want to continue doing that for this path?